proposals/0010-validating-resource-container-elements.md

```diff
@@ -0,0 +1,142 @@
+* Proposal: [0010](0010-validating-resource-container-elements.md)
+* Author(s): [Joshua Batista](https://github.com/bob80905)
+* Sponsor: TBD
+* Status: **Under Consideration**
+* Impacted Project(s): (LLVM)
+
+*During the review process, add the following fields as needed:*
+
+* PRs: [#NNNN](https://github.com/microsoft/DirectXShaderCompiler/pull/NNNN)
+* Issues: [#75676](https://github.com/llvm/llvm-project/issues/75676)
+
+## Introduction
+Resources are often used in HLSL, with various resource element types (RETs).
+For example:
+```
+RWBuffer<float> rwbuf: register(u0);
+```
+In this code, the RET is `float`, and the resource type is `RWBuffer`. The
+resource type is not a `RawBuffer` variant, and so there is a distinct set
+of rules that define valid RETs for this resource type.
+
+RETs for non-`RawBuffer` variants may include basic types (ints and uints of sizes 16
+and 32, as well as half, and float). Structs that contain fields of these basic
+types (where all fields in the struct have the same type) may also be RETs. 
+Structs that either have structs as fields or arrays of structs as fields may also be

```

Comment:
In addition, vectors and matrices of 4 elements or fewer are allowed both in and out of structs. I recognize that the matrix support is not part of this proposal, but it might be worth mentioning as part of the introduction potentially with a parenthetical or something that says it will be addressed separately.